streaming/src/main/scala/io/gearpump/streaming/StreamApplication.scala

```diff
@@ -144,6 +139,11 @@ object StreamApplication {
   def apply[T <: Processor[Task], P <: Partitioner] (name : String, dag: Graph[T, P], userConfig: UserConfig): StreamApplication = {
     import Processor._
 
+    val LOG = LogUtil.getLogger(this.getClass)
```

Comment:
OK. Because if there is loop in graph,  ``val indices = dag.topologicalOrderIterator.toList.zipWithIndex.toMap`` will be in infinite loop. And SAMOA task  get stucked.